### ISSUE TITLE
Add type conversion for parse String method

### ISSUE THREAD

**Issue by sdaschner** [eclipse-ee4j/jaxrs-api#879]
Chapter [Fields and Bean Properties](https://github.com/eclipse-ee4j/jaxrs-api/blob/master/jaxrs-spec/src/main/asciidoc/chapters/resources/_resource_field.adoc#fields-and-bean-properties) explains how implementations need to supports specific types for automatic field conversion.

In the Config JSR (382) back then, we had discussions around what to support and added a `parse(String)` / `parse(CharSequence)` method to this list (under point 4. in the list in the JAX-RS spec), which implicitly would enable automatic conversion for Java time types such as `LocalDate`.

If we add such a requirement (with lower priority to not break backwards compatibility) IMO we'd improve the user experience. WDYT?

**Comment by sdaschner** [comment]
FYI see https://github.com/eclipse/ConfigJSR/blob/master/spec/src/main/asciidoc/converters.asciidoc for the draft of the spec back then (now the JSR is withdrawn, however).

**Comment by mkarg** [comment]
The spec could mandate that each compliant implementation MUST provide an implementation of `ParamConverterProvider` with lowest `@Priority` (to be able to override it) that provides an instance of `ParamConverter<Object>` being able to auto-convert *all* classes providing `static parse(CharSequence)` (including all implementations of `CharSequence`). So there is no need for a special case in the spec, but just one enforced provider. In case #878 is accepted, the mandate could be reduced from `ParamConverterProvider` to `ParamConverter`. On the other hand, as one can author such a provider on his own, shall we really *enforce* it?

**Comment by chkal** [comment]
> The spec could mandate that each compliant implementation MUST provide an implementation of `ParamConverterProvider` with lowest `@Priority` (to be able to override it) that provides an instance of `ParamConverter<Object>` being able to auto-convert _all_ classes providing `static parse(CharSequence)` (including all implementations of `CharSequence`). So there is no need for a special case in the spec, but just one enforced provider.

I would also prefer an approach like this. However, to be consistent, it would require to use the same mechanism for `valueOf`, `String` constructors and the other types specified in section 3.2 of the spec.

> On the other hand, as one can author such a provider on his own, shall we really enforce it?

I would prefer to have out-of-the-box support for common types which are part of the JDK like `LocalDate`.

**Comment by sdaschner** [comment]
@mkarg You're right with the case that `String` implements `CharSequence`. In Config we made it explicit, since the impls have to do the extra work, and also we argued that users only really need the automatic conversion for anything except CharS xor String, and then the impls only have to check two signatures instead of investigating the param type, that was our motivation back then. But I'm fine with both.

> On the other hand, as one can author such a provider on his own, shall we really enforce it?

I would so too, for portability. Also the implementations have mechanisms for the other signatures as well already...
